src/puppet-puppeteer.ts

```diff
@@ -1427,51 +1427,89 @@ export class PuppetPuppeteer extends Puppet {
+      }
+      bufferData.push(tempBuffer)
+    }
+
+    async function getMediaId (buffer: Buffer, index: number) : Promise <string> {
+      const formData = {
+        chunk: index,
+        chunks,
+        filename: {
+          options: {
+            contentType,
+            filename,
+            size,
+          },
+          value: buffer,
         },
-        value: buffer,
-      },
-      id,
-      lastModifiedDate: Date().toString(),
-      mediatype,
-      name: filename,
-      pass_ticket: passTicket || '',
-      size,
-      type: contentType,
-      uploadmediarequest: JSON.stringify(uploadMediaRequest),
-      webwx_data_ticket: webwxDataTicket,
+        id,
+        lastModifiedDate: Date().toString(),
+        mediatype,
+        name: filename,
+        pass_ticket: passTicket || '',
+        size,
+        type: contentType,
+        uploadmediarequest: JSON.stringify(uploadMediaRequest),
+        webwx_data_ticket: webwxDataTicket,
+      }
+      try {
+        return await new Promise<string>((resolve, reject) => {
+          try {
+            request.post({
+              formData,
+              headers,
+              url: uploadMediaUrl + '?f=json',
+            }, (err, _, body) => {
+              if (err) {
+                reject(err)
+              } else {
+                let obj = body
+                if (typeof body !== 'object') {
+                  obj = JSON.parse(body)
+                }
+                resolve(obj.MediaId || '')
+              }
+            })
+          } catch (e) {
+            reject(e)
+          }
+        })
+      } catch (e) {
+        log.error('PuppetPuppeteer', 'uploadMedia() uploadMedia exception: %s', e.message)
+        throw new Error('uploadMedia err: ' + e.message)
+      }
+    }
+    let funcList = []
+    for (let i = 0; i < bufferData.length - 1; i++) {
+      funcList.push(await getMediaId(bufferData[i], i))
     }
+    // use promise.all() make the former upload of this buffer quickly
     let mediaId: string
     try {
-      mediaId = await new Promise<string>((resolve, reject) => {
-        try {
-          request.post({
-            formData,
-            headers,
-            url: uploadMediaUrl + '?f=json',
-          }, (err, _, body) => {
-            if (err) {
-              reject(err)
-            } else {
-              let obj = body
-              if (typeof body !== 'object') {
-                obj = JSON.parse(body)
-              }
-              resolve(obj.MediaId || '')
-            }
-          })
-        } catch (e) {
-          reject(e)
-        }
-      })
-    } catch (e) {
-      log.error('PuppetPuppeteer', 'uploadMedia() uploadMedia exception: %s', e.message)
-      throw new Error('uploadMedia err: ' + e.message)
+      await Promise.all(funcList)
+      const lastOne = bufferData.length - 1
+      mediaId = await getMediaId(bufferData[lastOne], lastOne)
+    } catch {
```

Comment:
I think we need to catch the error and log it here.

Reply:
Done